pdr_backend/contract/subscription.py

```diff
@@ -0,0 +1,137 @@
+from typing import List
```

Comment:
`contract/` directory is only for classes that are wrappers of smart contracts.

subscription.py is not that.

 To solve: move subscription.py to `lake/`.

Second: where is `test_subscription.py`?

Reply:
wrt `subscrptions.py`
we used to have `models/`  => `contracts/`
so `predictions.py` should also be moved to `lake/`

we also have `subgraph/` files referencing these objects
such as `subgraph_predictions.py`

should we perhaps move these to lake too?
i.e. all lake-related `subgraph/models/mocks/testing/` etc... inside the same module.

if so, then can you perhaps recommend a structure for this? Here a very simple, but perhaps redundant structure... I think your "line-in-the-sand" feedback would be useful.
`lake/`
`lake/models/`
`lake/subgraph/`

Reply:
Please no hierarchy for now. (Remember, every sub-directory you add also means another sub-sub directory for tests. It's a huge pain to manage. Eg even with predictoor/approach1 and 3, and trader/approach1 and 2. Super annoying to jump around. So, for now, keep the rest flat.)

"lake" is higher-level than subgraph.

Given that subgraph uses `subscriptions.py` (I missed that), then I recommend: move `subscription.py` to `subgraph/` too.

Reply:
Implemented test_subscription, moved `subscriptions.py` and `predictions.py`, expanded tests, and added logic such that we don't have extra calls being made to subgraph when they aren't needed.